Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm

```diff
@@ -662,18 +662,36 @@ static void registerLogHook()
     os_log_set_hook(OS_LOG_TYPE_DEFAULT, ^(os_log_type_t type, os_log_message_t msg) {
         if (msg->buffer_sz > 1024)
             return;
-        char* messageString = os_log_copy_message_string(msg);
-        String logString = String::fromUTF8(messageString);
-        free(messageString);
 
+        String logFormat = String::fromUTF8(msg->format);
         String logChannel = String::fromUTF8(msg->subsystem);
         String logCategory = String::fromUTF8(msg->category);
 
-        callOnMainRunLoop([logChannel = logChannel.isolatedCopy(), logCategory = logCategory.isolatedCopy(), logString = logString.isolatedCopy(), type] {
-            auto* connection = WebProcess::singleton().existingNetworkProcessConnection();
-            if (!connection)
-                return;
-            connection->connection().send(Messages::NetworkConnectionToWebProcess::LogOnBehalfOfWebContent(logChannel, logCategory, logString, type, getpid()), 0);
+        Vector<uint8_t> buffer;
+        buffer.resize(msg->buffer_sz);
+        memcpy(buffer.data(), msg->buffer, msg->buffer_sz);
+        Vector<uint8_t> privdata;
+        privdata.resize(msg->privdata_sz);
+        memcpy(privdata.data(), msg->privdata, msg->privdata_sz);
+
+        dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0), [logFormat = logFormat.isolatedCopy(), logChannel = logChannel.isolatedCopy(), logCategory = logCategory.isolatedCopy(), type = type, buffer = WTFMove(buffer), privdata = WTFMove(privdata)] {
```

Comment:
As far as I know (please double check), DISPATCH_QUEUE_PRIORITY_BACKGROUND is a *concurrent* dispatch queue. As a result, I don't think the ordering of our logging is guaranteed anymore?

Reply:
Makes sense, will check!

Reply:
https://developer.apple.com/library/archive/documentation/General/Conceptual/ConcurrencyProgrammingGuide/OperationQueues/OperationQueues.html#//apple_ref/doc/uid/TP40008091-CH102-SW5